### ISSUE TITLE
PCL is unable to find installed GTest

### ISSUE THREAD

**Issue by jasjuang** [PointCloudLibrary/pcl#3530]
As discussed in https://github.com/PointCloudLibrary/pcl/pull/3527, currently PCL requires a freshly cloned gtest in order to compile with it. There needs to be a way for PCL to link to installed gtest.

## Your Environment
* Operating System and version: Ubuntu 18.04
* Compiler: GCC 7
* PCL Version: Latest master

**Comment by taketwo** [comment]
> Currently PCL requires a freshly cloned gtest in order to compile with it

Nope, what makes you think so? On Ubuntu, it's sufficient to have `libgtest-dev` installed and PCL will pick it up.

> There needs to be a way for PCL to link to installed gtest.

GTest is installed only as source files. You can verify it yourself by calling `dpkg-query -L googletest` and observing that the package only contains source files, no libraries.

**Comment by jasjuang** [comment]
`libgtest-dev` must be a really old version before proper cmake is there. The gtest I am using is cloned from source, and then the usual `cmake ..`, `make`, `sudo make install`. There will be a `GTestConfig.cmake` installed after this which I don't think `libgtest-dev` has because most likely this version does not have the proper cmake `exports` yet. Can you give it a try and see whether PCL can detect manually built gtest?

**Comment by taketwo** [comment]
> `libgtest-dev` must be a really old version before proper cmake is there

Old or new, this is the reality on Ubuntu. Even 19.10 installs GTest without proper config file.

> Can you give it a try and see whether PCL can detect manually built gtest?

This depends on where you install this manually built Gtest. If you install it to `/usr` or to `/usr/local`, PCL will pick it up. If you install to a custom location, then it won't, unless you provide `GTEST_ROOT` variable indicating where PCL should look for your installation.

**Comment by jasjuang** [comment]
@taketwo Mine got installed to `/usr/lib/x86_64-linux-gnu/cmake/GTest/GTestConfig.cmake` and something as simple as 

```
find_package(GTest CONFIG REQUIRED)

add_executable(myTest ${MY_SRCS} ${MY_INCS})

target_link_libraries(myTest GTest::gtest_main)
```

can pick it up without additional configuration, but PCL can't. Where does your `GTestConfig.cmake` got installed into?

**Comment by kunaltyagi** [comment]
Ubuntu 16.04 is going EOL in a year.

@jasjuang Does your snippet work with `apt-get install` gtest for 17.10 (or 18.04)?

**Comment by jasjuang** [comment]
No, because proper cmake isn't in there yet for `apt-get install` gtest 17.10 nor 18.04. It makes sense for PCL to account for that but I think PCL should also work with the latest gtest with proper cmake.

**Comment by tkircher** [comment]
The problem seems to be that pcl/cmake/Modules/FindGTestSource.cmake is looking for src/gtest-all.cc, but a correct installation of googletest doesn't include that source file. Copying it there from the googletest source directory makes PCL happy.
